google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -457,8 +457,7 @@ private static String getAppEngineProjectIdFromMetadataServer() throws IOExcepti
 
   @InternalApi("Visible for testing")
   static boolean headerContainsMetadataFlavor(HttpResponse response) {
-    String metadataFlavorValue = response.getHeaders().getFirstHeaderStringValue("Metadata-Flavor");
-    return metadataFlavorValue != null && metadataFlavorValue.equals("Google");
+    return "Google".equals(response.getHeaders().getFirstHeaderStringValue("Metadata-Flavor"));
```

Comment:
I _personally_ think it's clearer to keep the actual header in its own variable, but /shrug. Matter of taste. :)